### PR TITLE
Support encoding JSON::Any as hash values with Crest::EnumeratedFlatParamsEncoder

### PR DESCRIPTION
**Pull request by cyangle**

### CODE REVIEW

File: src/crest/params_encoders/enumerated_flat_params_encoder.cr [hunk 1]
```diff
@@ -1,5 +1,10 @@
 module Crest
   class EnumeratedFlatParamsEncoder < Crest::ParamsEncoder
+    # ```
+    # Encoded array params start index, default 1
+    # ```
+    class_property array_start_index = 1
```
Comment (cyangle) [inline src/crest/params_encoders/enumerated_flat_params_encoder.cr:6]: I think it's better to have a new encoder class with array index starting with 0.

**Comment by cyangle** [comment]
It passed exiting test suite. I'll add new test cases for encoding JSON::Any values.

**Comment by cyangle** [comment]
@mamantoha Please review.

**Comment by mamantoha** [comment]
Hi @cyangle . Thanks for your contribution. Looks good to me 👍

**Comment by mamantoha** [comment]
@all-contributors please add @cyangle  for code

**Pull request merged**